test/test_datatype.jl

```diff
@@ -0,0 +1,48 @@
+using Base.Test
+using MPI
+
+MPI.Init()
+
+immutable Boundary
+  c::UInt16  # force some padding to be inserted
+  a::Int
+  b::UInt8
+end
```

Comment:
If `Tuple{Int,Float64}` has been created by a previous call to `type_create`, then it will.  I will make the function act recursively so the user does not have to add each component type before adding the parent type.